gnuradio-runtime/python/gnuradio/gr/bindings/block_gateway_python.cc

```diff
@@ -7,17 +7,200 @@
  *
  */
 
+#include <gnuradio/api.h>
+#include <gnuradio/block.h>
+#include <gnuradio/io_signature.h>
+#include <string>
+
 #include <pybind11/complex.h>
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
 namespace py = pybind11;
 
-#include <gnuradio/block_gateway.h>
+
+/*!
+ * The gateway block which performs all the magic.
+ *
+ * The gateway provides access to all the gr::block routines.
+ */
+typedef enum {
+    GW_BLOCK_GENERAL = 0,
+    GW_BLOCK_SYNC,
+    GW_BLOCK_DECIM,
+    GW_BLOCK_INTERP
+} gw_block_t;
+
+class block_gateway : virtual public gr::block
+{
+private:
+    py::handle _py_handle;
+
+public:
+    block_gateway(const py::handle& py_handle,
+                  const std::string& name,
+                  gr::io_signature::sptr in_sig,
+                  gr::io_signature::sptr out_sig)
+        : block(name, in_sig, out_sig)
```

Comment:
it must be an arcane facet of scoping rules that probably isn't worth exploiting, that you are in the `gr` namespace here. Go ahead and be explicit
```suggestion
        : gr::block(name, in_sig, out_sig)
```
(note that historically, the impl classes were also in the `gr` namespace; in moving this, it is no longer the case)